codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/integration/HttpProtocolGeneratorUtils.java

```diff
@@ -136,4 +142,16 @@ public static boolean isShapeWithResponseBindings(Model model, Shape shape, Http
         }
         return false;
     }
+
+    /**
+     * Returns a map of error names to their {@link ShapeId}.
+     *
+     * @param context the generation context
+     * @param operation the operation shape to retrieve errors for
+     * @return map of error names to {@link ShapeId}
+     */
+    public static Map<String, ShapeId> getOperationErrors(GenerationContext context, OperationShape operation) {
+        return new TreeMap<>(operation.getErrors().stream()
```

Comment:
nit: you can correct directly to a TreeMap using [Collections#toMap](https://docs.oracle.com/javase/8/docs/api/java/util/stream/Collectors.html#toMap-java.util.function.Function-java.util.function.Function-java.util.function.BinaryOperator-java.util.function.Supplier-)